cuda/matrix/csr_kernels.cu

```diff
@@ -57,6 +57,8 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "cuda/base/types.hpp"
 #include "cuda/components/atomic.cuh"
 #include "cuda/components/cooperative_groups.cuh"
+#include "cuda/components/intrinsics.cuh"
+#include "cuda/components/merging.cuh"
```

Comment:
Are you sure you need them in `csr_kernels.cu`, even though you did not add any code which uses it?

Reply:
> /ginkgo/common/matrix/csr_kernels.hpp.inc(516): error: identifier "group_merge" is undefined
> /ginkgo/common/matrix/csr_kernels.hpp.inc(556): error: identifier "group_merge" is undefined

Yes, it seems like nvcc tries to instantiate the templates even though we never instantiate them.

Reply:
I completely missed the `common/matrix/csr_kernels.hpp.inc` file (thought it was also a `merge` file) when looking at use cases, by bad.